File: app/src/main/java/in/kashewdevelopers/airdistance/HistoryDbHelper.java
package in.kashewdevelopers.airdistance;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class HistoryDbHelper extends SQLiteOpenHelper {

    // table details
    private static final String DB_NAME = "history";
    private static final int DB_VERSION = 1;


    // columns in table
    static String SRC_NAME = "src_name";
    static String SRC_LL = "src_lat_lng";
    static String DST_NAME = "dst_name";
    static String DST_LL = "dst_lat_lng";
    static String DISTANCE = "distance";
    static String HASH = "hash";
    private static String TIME = "timestamp";


    HistoryDbHelper(Context context) {
        super(context, DB_NAME, null, DB_VERSION);
    }


    @Override
    public void onCreate(SQLiteDatabase db) {
        String creatingDb = "CREATE TABLE " + DB_NAME + " (" +
                "_id INTEGER PRIMARY KEY AUTOINCREMENT," +
                SRC_NAME + " TEXT," +
                SRC_LL + " TEXT," +
                DST_NAME + " TEXT," +
                DST_LL + " TEXT," +
                DISTANCE + " TEXT," +
                HASH + " TEXT UNIQUE ON CONFLICT REPLACE," +
                TIME + " DATETIME DEFAULT CURRENT_TIMESTAMP);";

        db.execSQL(creatingDb);
    }


    void insert(SQLiteDatabase db, String srcName, String srcLL, String dstName, String dstLL, String distance) {
        String hash = srcName + srcLL + dstName + dstLL;
        hash = String.valueOf(hash.hashCode());

        ContentValues data = new ContentValues();
        data.put(SRC_NAME, srcName);
        data.put(SRC_LL, srcLL);
        data.put(DST_NAME, dstName);
        data.put(DST_LL, dstLL);
        data.put(HASH, hash);
        data.put(DISTANCE, distance);

        db.insert(DB_NAME, null, data);
    }


    Cursor get(SQLiteDatabase db) {
        return db.query(DB_NAME, null, null, null, null,
                null, TIME + " DESC");
    }


    void deleteAll(SQLiteDatabase db) {
        db.delete(DB_NAME, null, null);
    }


    void delete(SQLiteDatabase db, String hashCode) {
        db.delete(DB_NAME, HASH + " = ?", new String[]{hashCode});
    }


    @Override
    public void onUpgrade(SQLiteDatabase sqLiteDatabase, int i, int i1) {
    }

}

File: app/src/main/java/in/kashewdevelopers/airdistance/db/history/HistoryDbHelper.kt
package `in`.kashewdevelopers.airdistance.db.history

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class HistoryDbHelper(context: Context) : SQLiteOpenHelper(context, DB_NAME, null, DB_VERSION) {

    companion object {
        private const val DB_NAME = "history"
        private const val DB_VERSION = 1

        const val COLUMN_SRC_NAME = "src_name"
        const val COLUMN_SRC_LAT_LNG = "src_lat_lng"
        const val COLUMN_DST_NAME = "dst_name"
        const val COLUMN_DST_LAT_LNG = "dst_lat_lng"
        const val COLUMN_DISTANCE = "distance"
        const val COLUMN_HASH = "hash"
        const val COLUMN_TIME = "timestamp"
    }

    override fun onCreate(db: SQLiteDatabase?) {
        db ?: return

        val query = "CREATE TABLE $DB_NAME(" +
                "_id INTEGER PRIMARY KEY AUTOINCREMENT," +
                "$COLUMN_SRC_NAME TEXT," +
                "$COLUMN_SRC_LAT_LNG TEXT," +
                "$COLUMN_DST_NAME TEXT," +
                "$COLUMN_DST_LAT_LNG TEXT," +
                "$COLUMN_DISTANCE TEXT," +
                "$COLUMN_HASH TEXT UNIQUE ON CONFLICT REPLACE," +
                "$COLUMN_TIME DATETIME DEFAULT CURRENT_TIMESTAMP);"
        db.execSQL(query)
    }

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {}

    fun insert(db: SQLiteDatabase, srcName: String, srcLatLng: String, dstName: String, dstLatLng: String, distance: String) {
        val hash = "${(srcLatLng + dstLatLng).hashCode()}"

        val data = ContentValues()
        with(data) {
            put(COLUMN_SRC_NAME, srcName)
            put(COLUMN_SRC_LAT_LNG, srcLatLng)
            put(COLUMN_DST_NAME, dstName)
            put(COLUMN_DST_LAT_LNG, dstLatLng)
            put(COLUMN_HASH, hash)
            put(COLUMN_DISTANCE, distance)
        }

        db.insert(DB_NAME, null, data)
    }

    fun get(db: SQLiteDatabase): Cursor? {
        return db.query(DB_NAME, null, null, null,
                null, null, "$COLUMN_TIME DESC")
    }

    fun deleteAll(db: SQLiteDatabase) {
        db.delete(DB_NAME, null, null)
    }

    fun delete(db: SQLiteDatabase, hashCode: String) {
        db.delete(DB_NAME, "$COLUMN_HASH = ?", arrayOf(hashCode))
    }

    fun updateDestinationName(db: SQLiteDatabase, dstLatLng: String, dstName: String) {
        val data = ContentValues()
        data.put(COLUMN_DST_NAME, dstName)
        db.update(DB_NAME, data, "$COLUMN_DST_LAT_LNG = ?", arrayOf(dstLatLng))
    }

    fun updateSourceName(db: SQLiteDatabase, srcLatLng: String, srcName: String) {
        val data = ContentValues()
        data.put(COLUMN_SRC_NAME, srcName)
        db.update(DB_NAME, data, "$COLUMN_SRC_LAT_LNG = ?", arrayOf(srcLatLng))
    }

}